### PR TITLE
[Feature] Min max constraints

### PR DESCRIPTION
**Pull request by Nukesor**
This PR reworks ColumnConstraints and introduces Min+Max Constraints.

It also fixes a few edge-case bugs and introduces benchmarks

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,6 +20,8 @@ Features:
 - Pretty much every part of the table is customizable (borders, lines, padding, alignment).
 - Constraints on columns that allow some additional control over how to arrange content.
 - Cross plattform (Linux, macOS, Windows).
+- It's FAST!
+    Benchmarks show that a pretty big table with weird constraints is build in 4 _us_, which is `4*1e^6`.
```
Comment (dmaahs2017) [inline README.md:24]: Here's the unicode char for micro seconds
```suggestion
    Benchmarks show that a pretty big table with weird constraints is build in 4 _μs_, which is `4*1e^6`.
```

Comment (Nukesor) [reply README.md:24]: Thanks :)

**Comment by Nukesor** [comment]
@dmaahs2017 I would like your feedback regarding some naming.

What do you think about the new Constraint names. E.g. `Absolute, LowerBoundary, UpperBoundary, Boundaries`? 
`Min/Max` seemed kind of unfitting, especially if there's a "MinMax" Constraint.

Is boundary the correct term for the behavior at hand?

**Comment by dmaahs2017** [comment]
> @dmaahs2017 I would like your feedback regarding some naming.
> 
> What do you think about the new Constraint names. E.g. `Absolute, LowerBoundary, UpperBoundary, Boundaries`?
> `Min/Max` seemed kind of unfitting, especially if there's a "MinMax" Constraint.
> 
> Is boundary the correct term for the behavior at hand?

Considering how this is used in the `Table::set_constraints` fn:
```rust
table.set_constraints(vec![
    ColumnConstraint::LowerBoundary(Fixed(10)),
    ColumnConstrains::UpperBoundary(Fixed(8)),
    ColumnConstrain::Absolute(Fixed(10)),
]);
```

* It's not quite clear by the code that they effect the width.
* I'm not sure what the `Fixed` wrapper is for.
```

**Comment by Nukesor** [comment]
1. I'm ging to add some info to `Table::set_constraints` on what constraints are for. Proper docs are on the `ColumnConstraint` enum:

```
/// A Constraint can be added to a [columns](crate::Column).
///
/// They allow some control over Column widths as well as the dynamic arrangement process.
///
/// All percental boundaries will be ignored, if:
/// - the terminal width cannot be determined.
/// - you aren't doing dynamic content arrangement.
```

2. I see how the fixed wrapper is confusing.
    I did a `use comfy_table::Boundary::*`, which moves the Invariantes from `Boundary::*` directly into the namespace.
    It's basically a shortcut for `Boundary::Fixed` and `Boundary::Percentage`.
    -> Every Constraint can be either enforced with a fixed value or a relative percentage of the terminal width.

File: src/table.rs [hunk 2]
```diff
@@ -231,9 +231,14 @@ impl Table {
...
 
     /// Convenience method to set a [ColumnConstraint] for all columns at once.
+    /// Constraints are used to influence the way the columns will be arranged.
+    /// Check out their docs for more information.
     ///
-    /// Simply pass any iterable with ColumnConstraints.\
-    /// If more constraints are passed than there are columns, the superfluous constraints will be ignored.
+    /// **Attention:**
+    /// This function should be called after at least one row has been added to the table.
+    /// Before that, the columns won't initialized.
```
Comment (dmaahs2017) [inline src/table.rs:239]: Does it have to be a row, or could one set a header, then the constraints?

Comment (Nukesor) [reply src/table.rs:239]: Header would work as well. Good point :)

**Comment by Nukesor** [comment]
Thanks again for your Review @dmaahs2017  :)
Much appreciated

**Pull request merged**